File: coyote/models/ack_msg_range_in.py
# this file is @generated
import typing as t
from pydantic import Field

from ..internal.base_model import BaseModel


class AckMsgRangeIn(BaseModel):
    consumer_group: str = Field(alias="consumer_group")

    max_msg_id: int = Field(alias="max_msg_id")

    min_msg_id: t.Optional[int] = Field(default=None, alias="min_msg_id")

    name: str

File: src/models/ack_msg_range_in.rs
// this file is @generated
use serde::{Deserialize, Serialize};

#[non_exhaustive]
#[derive(Clone, Debug, Deserialize, Serialize)]
pub struct AckMsgRangeIn {
    pub consumer_group: String,

    pub max_msg_id: u64,

    #[serde(skip_serializing_if = "Option::is_none")]
    pub min_msg_id: Option<u64>,

    pub name: String,
}

impl AckMsgRangeIn {
    pub fn new(consumer_group: String, max_msg_id: u64, name: String) -> Self {
        Self {
            consumer_group,
            max_msg_id,
            min_msg_id: None,
            name,
        }
    }
}
